addrset/addrset.go

```diff
@@ -0,0 +1,106 @@
+package addrset
+
+import (
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// AddrState describes the state of a dialable addresses as seen by AutoNAT.
+type AddrState int
```

Comment:
can we keep these states this with the network reach-ability ones?

More generally, it seems that there are two axes here that this enum can maybe track separately: those being *source and *confidence in reachability